### DISCUSSION TITLE
Discussions REST API

### DISCUSSION THREAD

**Discussion by Bibo-Joshi** [community/community#4327] [Discussions] [answered]
Hi. I was wondering if there are any plans to add Discussions to the REST API in addition to making them available through the GraphQL API.
This already came up in #43 (https://github.com/github/feedback/discussions/43#discussioncomment-495804, https://github.com/github/feedback/discussions/43#discussioncomment-625786, https://github.com/github/feedback/discussions/43#discussioncomment-796087, https://github.com/github/feedback/discussions/43#discussioncomment-914205), but that thread has quite a number of comments by now so I guess it just drowned, which is why I'm opening a new thread here :)

**Comment by mistic100** [comment]
Hoping for a REST API too. I don't feel needing to use a whole new library just to list the announcements on my documentation home.

**Reply by mistic100** [reply]
If someone has the same need and is using Netlify, you can use [Netlify Functions](https://docs.netlify.com/functions/build-with-javascript/) to setup a proxy to the GraphQL API

function : https://github.com/mistic100/Photo-Sphere-Viewer/blob/dev/docs/.netlify/functions/announcements.js  
call : https://github.com/mistic100/Photo-Sphere-Viewer/blob/dev/docs/.vuepress/components/Announcements.vue#L36

Just need to register GH_TOKEN in your Netlify env variables

**Reply by CodingwithFawad** [reply]
Thank you!! :

**Comment by Katsute** [comment, marked as answer]
I asked them in the recent Discussions AMA. <https://github.community/t/community-ama-let-s-discuss-discussions/205558/6>

They have no plans on adding a REST API for this service, or any future service; they are switching to GraphQL.

**Reply by mistic100** [reply]
that's a bummer

**Reply by Poolitzer** [reply]
Thanks for stating this!

**Reply by CodingwithFawad** [reply]
Thank you!! :

**Comment by mistic100** [comment]
And of course the GraphQL API requires token authentication even for read-only queries. Thus requiring to setup a server for the least basic operation, what a waste of resources.

**Reply by ArjixWasTaken** [reply]
well, you could always just use a repo that has a workflow that generates a json file with all the data you want
but that is too much effort ig

**Reply by CodingwithFawad** [reply]
Thank you!! :
